### DISCUSSION TITLE
Javas missing unsigned byte

### DISCUSSION THREAD

**Discussion by mores** [Pi4J/pi4j#379] [General]
Is there a pi4j standard or recommendation how to deal with javas lack of an unsigned byte ?
 - create variables as short then cast them to byte ?
 `short myValue = 0xff;`
`byte[] data = new byte[1];`
`data[0] = (byte)myValue;`

- Or utilize a third part library like: https://github.com/jOOQ/jOOU
- Or is there a Utility class somewhere in pi4j that could be used ?

**Comment by FDelporte** [comment]
Thanks @mores, actually a very good question, but can't answer it directly...
This calls for some experiments and a (blog) post... 
I put it on my agenda for "asap".

**Comment by eitch** [comment]
Hi @mores I usually write the values as hex, e.g. 0xff as you do, then case to byte, so this would be a one liner for me:

    byte[] data = new byte[] { (int) 0xff};

I understand it can be cumbersome, but then i usually use constants for special values and then don't need to think about it anymore.

If you want such utility functions then you can send a PR for it, but i personally don't want to add another dependency to the pi4j project for just a few simple helper functions.

**Comment by FDelporte** [comment]
Hi, I blogged about this, and created a video:
https://webtechie.be/post/2024-09-26-java-bits-bytes-short-int-long-signed-unsigned/

This helper method in Java can be very handy: `Byte.toUnsignedInt(b)`
